### ISSUE TITLE
`no-unnecessary-duplication rule` false-positive detections

### ISSUE THREAD

**Issue by igorkamyshev** [igorkamyshev/eslint-plugin-effector#21]
The rule triggers on this code:
```ts
guard({
  source: calendarAviaFormService.$aviaForm,
  clock: calendarAviaFormService.submitted,
  filter: (state) =>
    Boolean(state && isValidDirection(state.origin, state.destination)),
  target: drawAttention,
});
```
It must not.

**Comment by igorkamyshev** [comment]
Here too:

```ts
guard({
  source: promocodeModel.$verifyError,
  clock: [promocodeChanged, promocodeInputBlurred],
  filter: Boolean,
  target: promocodeModel.verifyErrorReset,
});
```

**Comment by igorkamyshev** [comment]
One more case:
```ts
guard({
  source: [
    aviaFormService.$aviaForm,
    $backendFilters,
    currencyService.$currency,
  ],
  clock: [$backendFilters, pricesPublicApi.allPricesShown],
  filter: $isAviaFormValidForPrices,
  target: requestLatestPricesFx,
});
```
